lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -77,25 +75,18 @@ newTransactionLayer
         )
     => Hash "Genesis"
     -> TransactionLayer t k
-newTransactionLayer (Hash block0H) = TransactionLayer
+newTransactionLayer block0H = TransactionLayer
     { mkStdTx = \keyFrom rnps outs -> do
-        -- NOTE
-        -- For signing, we need to embed a hash of the transaction data
-        -- without the witnesses (since we don't yet have them!). In this sense,
-        -- this is a transaction id as Byron nodes or the http-bridge
-        -- defines them.
-        let inps = fmap (second coin) rnps
-        wits <- forM rnps $ \(_, TxOut addr _) -> do
-            xprv <- maybeToRight (ErrKeyNotFoundForAddress addr) (keyFrom addr)
-            let payload = block0H <> getHash (signData inps outs)
-            pure $ mkTxWitness (fst xprv) (sign payload xprv)
-        let tx = Tx
-                { txId = fragmentId inps outs wits
-                , resolvedInputs = inps
-                , outputs = outs
-                }
-        let binary = signedTransactionFragment tx wits
-        return (tx, binary)
+        (fragId, frag) <- constructTransactionFragment
+            rnps
+            outs
+            (\addr -> do
+                key@(xprv, _pass)
+                    <- maybeToRight (ErrKeyNotFoundForAddress addr) (keyFrom addr)
+                return $ \txHash -> mkTxWitness xprv txHash block0H (`sign` key)
```

Comment:
Could probably be made a little bit nicer by re-organizing the order of arguments in `mkTxWitness`.